db/migrate/20150226095809_add_status_enums_to_orphan.rb

```diff
@@ -0,0 +1,30 @@
+class AddStatusEnumsToOrphan < ActiveRecord::Migration
+  def up
+    add_column :orphans, :status, :integer, default: 0
+    add_column :orphans, :sponsorship_status, :integer, default: 0
+
+    set_status
+    set_sponsorship_status
+  end
+
+  def down
+    remove_column :orphans, :status
+    remove_column :orphans, :sponsorship_status
+  end
+
+  private
+
+  def set_status
+    Orphan.all.each do |orphan|
+      status_code = orphan.orphan_status.code
+      orphan.update_column(:status, ( status_code - 1 ))
+    end
+  end
+
+  def set_sponsorship_status
+    Orphan.all.each do |orphan|
+      status_code = orphan.orphan_sponsorship_status.code
+      orphan.update_column(:sponsorship_status, ( status_code - 1 ))
+    end
+  end
+end
```

Comment:
First, I'm really not sure how to do that programmatically. Even the relatively straight-forward
```ruby
  def migration_has_been_run?(version)
    query = "SELECT version FROM schema_migrations WHERE version = '#{version}'"
    ActiveRecord::Base.connection.execute(query).any?
  end
```
is very brittle and eventually begins to return `false` for reasons I don't understand. And that's when the migration in question is the last one.

Secondly, this and all other migrations will only run once in production since we don't have the luxury of resetting or dropping the db. In development and test, all migrations run before the db is seeded, so the conversion of existing data into a new format is not an issue.